profiles/common/modules/features/nexteuropa_editorial/includes/nexteuropa_editorial.tracking.inc

```diff
@@ -0,0 +1,429 @@
+  // are not supported yet.
+  foreach ($ent_field_instances as $field_name => $ent_field_instance) {
+    $field_info = field_info_field($field_name);
+    $widget_type = _nexteuropa_editorial_field_get_widget_type($ent_field_instance);
+    // To cover the case where a field is not translatable while
+    // the entity is.
+    $field_languages = array(LANGUAGE_NONE);
+    if ($field_info['translatable']) {
+      $field_languages = $available_transl_langs;
+    }
+
+    // Detecting tracked changes in entity by screening fields.
+    if (in_array($widget_type, $screened_field_types)) {
+      $field_settings = $ent_field_instance['settings'];
+      if ($field_settings['text_processing']) {
+        foreach ($field_languages as $translation_lang) {
+          $values = field_get_items($type, $entity, $field_name, $translation_lang);
+          if ($values) {
+            foreach ($values as $value) {
+              if (preg_match('/<span[^>]+class="[^"]*ice-[^>]+>/', $value['value'])) {
+                $entity->tracking_activate[$translation_lang] = $translation_lang;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Entity validation callback.
+ *
+ * It checks if the submitted form contains tracking changes.
+ * If not, it checks if the previous entity revision or translations
+ * contained some.
+ * Then, depending on settings, it blocks the form submitting.
+ *
+ * @param array $form
+ *   The submitted form.
+ * @param array $form_state
+ *   The form state of the submitted form.
+ */
+function _nexteuropa_editorial_tracking_entity_validate($form, &$form_state) {
+  $values = $form_state['values'];
+
+  // So far, we only support "node" entity type.
+  if (isset($form_state['node'])) {
+    $entity = $form_state['node'];
+    $current_wbm_state = $values['workbench_moderation_state_new'];
+
+    // Get current entity language.
+    $current_ent_lang = $values['language'];
+    if (isset($form_state['entity_translation'])) {
+      // Entity translation case.
+      $current_ent_lang = $form_state['entity_translation']['form_langcode'];
+    }
+
+    $tracking_to_check = _nexteuropa_editorial_tracking_saving_to_block($entity, $current_wbm_state);
+    // Generates errors only if required by WYSIWYG tracking workflow settings.
+    if ($tracking_to_check) {
+      $tracking_activate = $values['tracking_activate'];
+
+      // Unset current language because current form could be emptied of
+      // its tracked changes.
+      unset($tracking_activate[$current_ent_lang]);
+
+      // Treating current revision.
+      $tracking_detected = FALSE;
+      if (isset($form_state['tracked_changed_detected'])) {
+        $tracking_detected = $form_state['tracked_changed_detected'];
+      }
+
+      if ($tracking_detected) {
+        foreach ($form_state['tracked_changed_detected'] as $field_path) {
+          form_set_error($field_path, t('The field contains tracked changes that may not be saved.'));
```

Comment:
It would be good to pass field name to the message. Important for cases where you will have more WYSIWYG fields instances.